saleor/order/utils.py

```diff
@@ -367,9 +384,35 @@ def add_gift_card_to_order(order, gift_card, total_price_left):
     return total_price_left
 
 
-def change_order_line_quantity(user, line, old_quantity, new_quantity):
+def decrease_allocations(lines_info):
+    """Decreate allocations for provided order lines."""
+    if not get_order_lines_with_track_inventory(lines_info):
+        return
+    decrease_stock(lines_info, update_stocks=False)
+
+
+def _update_allocations_for_line(
+    line_info: OrderLineData, old_quantity: int, new_quantity: int
+):
+    if old_quantity == new_quantity:
+        return
+
+    if not get_order_lines_with_track_inventory([line_info]):
+        return
+
+    if old_quantity < new_quantity:
+        increase_allocation(line_info, new_quantity - old_quantity)
+    else:
+        line_info.quantity = old_quantity - new_quantity
+        decrease_allocations([line_info])
```

Comment:
I'm wondering if we could unify somehow `increase_allocation` and `descrease_allocations` - one is accepting a single line, while the other is accepting a list of lines. One updates `line_info.quantity` while the other does not and we do it in line 406 manually. Shouldn't `decrease_allocations` be moved to `warehouse/management.py` where other similar functions live?

Reply:
Now both accept the lines and were moved to management, should be much cleaner now.